### PR TITLE
chore: use Jest projects internally

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Uses [Jest projects](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig) internally to enable per-project configuration. 

We don't have any different settings across different projects yet, but in theory we could now.

To test locally, `yarn test`.

To test in CI, see that the [build-test flow in CI](https://app.circleci.com/pipelines/github/Codecademy/client-modules/16449/workflows/7448de81-3114-4280-a50d-4c3d2944c6f8) does run all tests across the jobs.

### PR Checklist

- ~[ ] Related to designs:~
- [x] Related to JIRA ticket: WEB-548
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: script/jest/rtl-setup.js [hunk 1]
```diff
@@ -0,0 +1 @@
+require('@testing-library/jest-dom');
```
Comment (JoshuaKGoldberg) [inline script/jest/rtl-setup.js:1]: I couldn't figure out how to get Babel to work outside of the package directories and eventually gave up. I'm not too hurt by this one line of CommonJS.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.9.4-alpha.d23da5.0<br/>@codecademy/gamut-illustrations@0.17.21-alpha.d23da5.0<br/>@codecademy/gamut-kit@0.5.64-alpha.d23da5.0<br/>@codecademy/gamut-labs@19.24.8-alpha.d23da5.0<br/>@codecademy/gamut-patterns@0.5.9-alpha.d23da5.0<br/>@codecademy/gamut-styles@11.9.3-alpha.d23da5.0<br/>@codecademy/gamut-system@0.7.4-alpha.d23da5.0<br/>@codecademy/gamut-tests@2.4.24-alpha.d23da5.0<br/>@codecademy/gamut@37.1.1-alpha.d23da5.0<br/>@codecademy/konamimojisplosion@0.2.55-alpha.d23da5.0<br/>@codecademy/macros@2.1.2-alpha.d23da5.0<br/>@codecademy/markdown-overrides@0.5.68-alpha.d23da5.0<br/>@codecademy/monaco@0.3.13-alpha.d23da5.0<br/>@codecademy/styleguide@44.1.4-alpha.d23da5.0<br/>@codecademy/tracking@0.13.2-alpha.d23da5.0<br/>@codecademy/variance@0.17.5-alpha.d23da5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://613772a5512f0fafcb178b01--gamut-preview.netlify.app>https://613772a5512f0fafcb178b01--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/613772a5512f0fafcb178b01>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]
ooo this is great!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.10.1-alpha.cb7aca.0<br/>@codecademy/gamut-illustrations@0.18.1-alpha.cb7aca.0<br/>@codecademy/gamut-kit@0.5.74-alpha.cb7aca.0<br/>@codecademy/gamut-labs@19.26.3-alpha.cb7aca.0<br/>@codecademy/gamut-patterns@0.6.1-alpha.cb7aca.0<br/>@codecademy/gamut-styles@11.10.1-alpha.cb7aca.0<br/>@codecademy/gamut-system@0.8.1-alpha.cb7aca.0<br/>@codecademy/gamut-tests@2.6.1-alpha.cb7aca.0<br/>@codecademy/gamut@37.3.2-alpha.cb7aca.0<br/>@codecademy/konamimojisplosion@0.3.1-alpha.cb7aca.0<br/>@codecademy/macros@2.2.1-alpha.cb7aca.0<br/>@codecademy/markdown-overrides@0.6.1-alpha.cb7aca.0<br/>@codecademy/monaco@0.4.2-alpha.cb7aca.0<br/>@codecademy/styleguide@44.3.3-alpha.cb7aca.0<br/>@codecademy/tracking@0.14.2-alpha.cb7aca.0<br/>@codecademy/variance@0.18.1-alpha.cb7aca.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://613f73f4ed232a198d2fb85b--gamut-preview.netlify.app>https://613f73f4ed232a198d2fb85b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/613f73f4ed232a198d2fb85b>Deploy Logs</a></p>

**Pull request merged**